RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,302 @@
+
+### Abstract
+
+The Tari communication network consists of a large number of nodes that maintained peer connections between each other.
+These nodes forward and propagate encrypted and unencrypted data messages through the network such as joining requests, discovery requests, [transaction]s and completed [block]s.
+Network clients, not responsible for maintaining the network, are able to create ad hoc connections with nodes on the network to perform joining and discovery requests.
+The majority of communication between clients and nodes will be performed using direct Peer-to-peer (P2P) communication once the discovery process was used to obtain the online communication addresses of peers.
+Where possible the efficient Kademlia based directed forwarding of encrypted data messages can be used to perform quick node discovery and joining of clients and nodes on the Tari communication network. 
+Where messages are of importance to a wide variety of entities on the network, Gossip protocol based message propagation can be performed to distribute the message to the entire network.
+
+### Overview
+
+The Tari communication network is a variant of a [Kademlia](https://en.wikipedia.org/wiki/Kademlia) network that allows for fast discovery of nodes, with an added ability to perform Gossip protocol based broadcasting of data messages to the entire network.
+The majority of the communication required on the [Base Layer] and [Digital Asset Network] (DAN) will be performed via direct P2P communication between known clients and nodes.
+Alternatively, the Tari communication network can be used for broadcasting joining requests, discovery requests and propagating data messages such as completed blocks, transactions and data messages that are of interest to a large part of the Tari communication network. 
+
+The Tari communication network consists of a number of different entities that need to communicate in a distributed and ad-hoc manner. 
+The primary entities that need to communicate are Validator Nodes (VN), [Base Node]s (BN), [Wallet]s (W) and Token Wallet (TW).
+Here are some examples of different communication tasks that need to be performed by these entities on the Tari Communication network:
+- Base Nodes on the Base Layer need to propagate completed blocks and transactions to other Base Nodes using Gossip protocol based broadcasting.
+- Validator Nodes need to efficiently discover other Validator Nodes in the process of forming Validator Node committees. 
+- Wallets need to communicate and negotiate with other Wallets to create transactions. They also need the ability to submit transactions to the [mempool] of Base Nodes.
+- Token Wallets need to communicate with Validator Node [committee]s and other Token Wallets to construct and send DAN instructions.
+
+Here is an overview communication matrix that show which source entities need to communicate with other destination entities on the Tari Communication network:
```

Comment:
I think this table should be added to the MUST requirement category.

Reply:
I'm not sure this should be a "MUST". Technically, any entity can speak to any other entity on the network, but it could be implemented that any comms that is "No" in this table could be ignored?

Reply:
Let us consider SHOULD....

Reply:
Change applied.